src/main/java/com/amazonaws/sagemaker/controller/ServingController.java

```diff
@@ -148,6 +150,54 @@ public ResponseEntity returnBatchExecutionParameter() throws JsonProcessingExcep
         }
     }
 
+    /**
+     * Implements the invocations POST API for application/jsonlines input
+     *
+     * @param jsonLines, lines of json values
+     * @param accept, accept parameter from request
+     * @return ResponseEntity with body as the expected payload JSON & proper statuscode based on the input
+     */
+    @RequestMapping(path = "/invocations", method = POST, consumes = AdditionalMediaType.APPLICATION_JSONLINES_VALUE)
+    public ResponseEntity<String> transformRequestJsonLines(@RequestBody final byte[] jsonLines,
+                                                            @RequestHeader(value = HttpHeaders.ACCEPT, required = false) final String accept) {
+        if (jsonLines == null) {
+            LOG.error("Input passed to the request is empty");
+            return ResponseEntity.noContent().build();
+        }
+        try {
+            final String acceptVal = this.retrieveAndVerifyAccept(accept);
+            final DataSchema schema = this.retrieveAndVerifySchema(null, mapper);
+            final String jsonStringLine = new String(jsonLines);
+
+            // Map list of inputs to DataList object
+            final SageMakerDataListObject sro = mapper.readValue(jsonStringLine, SageMakerDataListObject.class);
+            List<List<Object>> inputDatas = sro.getData();
+            List<ResponseEntity<String>> responseList = Lists.newArrayList();
+
+            // Process each input separately and add response to a list
+            final int inputDatasSize = inputDatas.size();
```

Comment:
I guess you could potentially combine this block and the next into a separate method called processInput and then just add the output from that to the final response list. That way - it might be more readable?

Reply:
Yes, we may create a new method called processInputDataForMutiplelinesInput since we already had a method called processInputData.